react-mutations/my-app/src/App.tsx

```diff
@@ -20,17 +20,15 @@ export function App() {
   }
 
   function handleUpdate(): void {
-    const updatedPoke = {
-      number: '001',
-      name: 'Foggy',
-    };
-    setPokedex([...pokedex, updatedPoke]);
+    const updatePokedex = pokedex.map((pokemon) =>
+      pokemon.number === '039' ? { ...pokemon, name: 'Alfons' } : pokemon
+    );
+    setPokedex(updatePokedex);
   }
-
   function handleRemove(): void {
     const deletedItems = [...pokedex];
-    deletedItems.splice(deletedItems.length - 1, 1);
-    setPokedex(deletedItems);
+    const item = deletedItems.filter((poke) => poke.number !== '004');
```

Comment:
This works, but it's easiest to just use `pokedex` instead of `deletedItems`. The function would then look like:
```
const items = pokedex.filter((poke) => poke.number !== '004');
setPokedex(items);
```